staging/src/k8s.io/apimachinery/pkg/api/errors/errors.go

```diff
@@ -757,7 +757,8 @@ func IsRequestEntityTooLargeError(err error) bool {
 // and may be the result of another HTTP actor.
 // It supports wrapped errors and returns false when the error is nil.
 func IsUnexpectedServerError(err error) bool {
-	if status := APIStatus(nil); errors.As(err, &status) && status.Status().Details != nil {
+	status, ok := err.(APIStatus)
```

Comment:
Are these functions tested? It seems that this changes the type of `status` and I suspect they don't do exactly the same thing anymore?

Reply:
I guess APIStatus is an interface type, let me rethink this